### PR TITLE
Chart messgage 1355

### PR DESCRIPTION
**Pull request by shivani170**
# Description
The reason for the degraded status is not visible on devtron. In my case, it's the apiversion which was missing. which is visible on argocd but not on devtron app details page.
Screenshot 2022-03-28 at 6 38 20 AM

Fixes https://github.com/devtron-labs/devtron/issues/1355

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
- [x] When the app status is `Degraded` 
- [x] With the character length of the error less than 120
- [x] With the character length of the error more than 120

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx [hunk 1]
```diff
@@ -1,131 +1,178 @@
...
+            (agg, curr) => {
+                agg.set(`${curr.kind}/${curr.name}`, curr)
+                return agg
+            },
+            new Map(),
+        )
+        setNodeStatusMap(stats)
+    }, [appStreamData])
+
+    function getNodeMessage(kind, name) {
```
Comment (vivek-devtron) [inline src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx:35]: Please provide types

File: src/components/v2/appDetails/sourceInfo/environmentStatus/EnvironmentStatus.component.tsx [hunk 2]
```diff
@@ -13,7 +13,7 @@ import { Link } from 'react-router-dom';
 import { useRouteMatch, useHistory } from 'react-router';
 import Tippy from '@tippyjs/react';
 
-function EnvironmentStatusComponent() {
+function EnvironmentStatusComponent({appStreamData}) {
```
Comment (vivek-devtron) [inline src/components/v2/appDetails/sourceInfo/environmentStatus/EnvironmentStatus.component.tsx:16]: Provide Type if possible

File: src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx [hunk 3]
```diff
@@ -1,131 +1,178 @@
...
-//     if (nodeStatusMap && nodeStatusMap.has(`${kind}/${name}`)) {
-//         const { status, message } = nodeStatusMap.get(`${kind}/${name}`);
-//         if (status === 'SyncFailed') return 'Unable to apply changes: ' + message;
-//     }
-//     return '';
-// }
-
-function AppStatusDetailModal({ close }) {
-    const _appDetails = IndexStore.getAppDetails();
+function AppStatusDetailModal({ close, appStreamData }) {
```
Comment (sohel-rp) [inline src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx:9]: Please provide types

File: src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx (hunk 1 above)
Comment (shivani170) [reply src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx:35]: done

File: src/components/v2/appDetails/sourceInfo/environmentStatus/EnvironmentStatus.component.tsx (hunk 2 above)
Comment (shivani170) [reply src/components/v2/appDetails/sourceInfo/environmentStatus/EnvironmentStatus.component.tsx:16]: done

File: src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx (hunk 3 above)
Comment (shivani170) [reply src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx:9]: done

**Review by vivek-devtron — APPROVED** [review]

**Review by sohel-rp — APPROVED** [review]

**Review by sohel-rp — APPROVED** [review]

**Review by pghildiyal — APPROVED** [review]

**Pull request merged**